src/stores/depthChartStore.ts

```diff
@@ -132,28 +188,157 @@ class DepthChartStore extends BaseStore {
     }
   };
 
-  @action
-  setWidth = (width: number) => {
-    this.width = width;
+  reduceBidsArray = (bids: Order[]) => {
+    const maximum = Math.max(...bids.map(b => b.price));
+    const lowerBound =
+      maximum - maximum * this.multiplers[this.spanMultiplierIdx];
+    const filteredBids = bids.filter(bid => bid.price > lowerBound);
+    return filteredBids.length ? filteredBids : bids.slice(0, 1);
   };
 
-  @action
-  setHeight = (height: number) => {
-    this.height = height;
+  reduceAsksArray = (asks: Order[]) => {
+    const minimum = Math.min(...asks.map(a => a.price));
+    const upperBound =
+      minimum + minimum * this.multiplers[this.spanMultiplierIdx];
+    const filteredAsks = asks.filter(ask => ask.price < upperBound);
+    return filteredAsks.length
+      ? filteredAsks
+      : asks.slice(asks.length - 1, asks.length);
   };
 
-  @computed
-  get isMaxZoom(): boolean {
+  updateAsks = (asks: Order[]) => (this.asks = asks);
+  updateBids = (bids: Order[]) => (this.bids = bids);
+
+  mid = async () => await this.rootStore.orderBookStore.mid();
+
+  spread = async () => {
+    const bestAsk = await this.rootStore.orderBookStore.getBestAsk();
+    const bestBid = await this.rootStore.orderBookStore.getBestBid();
+    return ((bestAsk - bestBid) / bestAsk) * 100;
+  };
+
+  calculateExactPrice = (area: DepthArea, index: number): number => {
+    return area === DepthArea.Bid
+      ? this.calculateExactBidPrice(index)
+      : this.calculateExactAskPrice(index);
+  };
+
+  calculateExactBidPrice = (index: number) => {
+    return take(this.getBids.length - index, this.getBids).reduce(
+      (sum: number, curr: Order) => sum + curr.volume * curr.price,
+      0
+    );
+  };
+
+  calculateExactAskPrice = (index: number) => {
+    return take(index, reverse(this.getAsks)).reduce(
+      (sum: number, curr: Order) => sum + curr.volume * curr.price,
```

Comment:
Reducer can be placed in other function, you can also use it in previous function